Adding tests.

MatthieuDartiailh opened the pull request:
I started adding tests. For the time being only the most basic ones are implemented.

There is actually no bug in Python 3... I was testing locally against an old version.

sccolbert commented:
Should I hold off on this until the commented-out stuff is handled?

MatthieuDartiailh commented:
Yes I hope to complete the tests today. I will let you know when this is ready.

MatthieuDartiailh commented:
I added tests for Expression, Constraint and Solver. The Solver does not do anything super interesting as I was a bit short on time and ideas (that is also why I left the test sanity in place). All the Python API is tested, which, with respect to the recent updates, is the most important.

MatthieuDartiailh commented:
I added another tests checking that constraint strength are respected. I think that we can remove safely test_kiwi_solver_basic_sanity now.
This is good to go as far as I am concerned.

MatthieuDartiailh commented:
@sccolbert are you satisfied with those tests ?

MatthieuDartiailh commented:
ping @sccolbert

MatthieuDartiailh commented:
@sccolbert any hope you could give this a quick look ?

Qix- commented:
*poke*

MatthieuDartiailh commented:
Ping

jonathanrocher commented:
Adding my own ping to the list @sccolbert .

MatthieuDartiailh commented:
Thanks ! Hope this will work !

jonathanrocher commented:
Thank *you* for the work you put in this repo! We need to make it work, one way or another because there are projects depending on this package (notably the Enthought Tool Suite which I am using). I haven't looked at the kiwi code to be able to review your PR unfortunately. Hope @sccolbert can way in on the future of this repo...

MatthieuDartiailh commented:
ping @sccolbert

MatthieuDartiailh commented:
I quickly reviewed what I did a year ago and concluded that the sanity checks were covered by the new check and I removed them. I also added support for creating anonymous variable in Python since the cpp API now allows it. In the absence of comments I will squash and merge tomorrow morning (UTC +1).

sccolbert commented:
:+1: :shipit:

QuLogic commented on setup.py:
```diff
@@ -49,7 +49,7 @@ def build_extensions(self):
     description='A fast implementation of the Cassowary constraint solver',
     long_description=open('README.rst').read(),
-    install_requires=['distribute'],
+    install_requires=['setuptools'],
```
`install_requires` is for [runtime deps](https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires). Since this is a shared library, is `setuptools` really required at runtime?
